### PR TITLE
fix(core): validate correctly formatted data-colorbox-opts value

### LINKED ISSUE #6104: Throw error if invalid JSON for colorbox config
If the raw `data-colorbox-opts` attribute exists and `$element.data('colorboxOpts')` is null, throw an error suggesting the use of `json_encode()`.

See https://github.com/Elgg/Elgg/issues/6103#issuecomment-25415072

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #6104 
- [ ] blocked on #7982

### CODE REVIEW

File: engine/lib/output.php [hunk 1]
```diff
@@ -153,6 +153,12 @@ function elgg_format_attributes(array $attrs = array()) {
 
 		unset($attrs['js']);
 	}
+	
+	if (isset($attrs['data-colorbox-opts'])) {
+		if (!@json_decode($attrs['data-colorbox-opts'])) {
```
Comment (mrclay) [inline engine/lib/output.php]: Doesn't the user already get some sort of error indication in console? I don't like the idea of JSON validating input to views, and especially not in this function.

Comment (jdalsem) [inline engine/lib/output.php]: no error in js console, this was the only nice place to do this in php

Comment (mrclay) [inline engine/lib/output.php]: Seems like we could detect this problem client-side by sniffing the raw attribute content. I just really don't want this function to start taking on validation of business logic.

Comment (jdalsem) [inline engine/lib/output.php]: that was my first plan, but it is only doable "onclick" and there is no notice of debug level, so you can only limit it to admin only. We have no good debug reporting capabilities client side.

**Comment by mrclay** [comment]
I would rather see if we can solve #7982 first.

**Comment by mrclay** [comment]
**Note:** This was closed simply because the 1.x branch was retired. Consider sending a new PR for the 2.x branch.

**Pull request closed without merging**